drivers/can/can_common.c

```diff
@@ -105,9 +105,17 @@ static int update_sampling_pnt(uint32_t ts, uint32_t sp, struct can_timing *res,
 		}
 	}
 
+	res->phase_seg2 = ts2;
+
+	/* Attempt to distribute ts1 evenly between prop_seq and phase_seg1 */
 	res->prop_seg = CLAMP(ts1 / 2, min->prop_seg, max->prop_seg);
 	res->phase_seg1 = ts1 - res->prop_seg;
-	res->phase_seg2 = ts2;
+
+	if (res->phase_seg1 > max->phase_seg1) {
```

Comment:
Theoretically if `phase_seg1` would have a min value greater than the min value of `prop_seg` it could still lead to an invalid result for `phase_seg1`.

Reply:
@str4t0m You are right, thank you for spotting this. I had somehow convinced myself this check was already covered by the clamping of the propagation segment, but clearly it isn't. I have update the code to also check the minimum boundary of `phase_seg1`.